Update: Display layer view state (#921)
Esri/arcgis-runtime-samples-ios: arcgis-ios-sdk-samples/Maps/Display layer view state/README.md

# Display layer view state

> Determine if a layer is currently being viewed.

Author: Nice catch! 😓

![Image of display layer view state](display-layer-view-state.png)

## Use case

The view status includes information on the loading state of layers and whether layers are visible at a given scale. You might change how a layer is displayed in a layer list to communicate whether it is being viewed in the map. For example, you could show a loading spinner next to its name when the view status is `loading`, gray out the name when `notVisible` or `outOfScale`, show the name normally when `active`, or with an error icon when the state is `warning` or `error`.

## How to use the sample

When the feature layer is loaded, pan and zoom around the map. Note how the `AGSLayerViewState` flags change; for example, `outOfScale` becomes true when the map is scaled outside of the layer's min and max scale range. Tap the toggle to hide the layer and observe the view state change to `notVisible`.

> You can toggle on airplane mode or disconnect from network and pan around the map to see layers display the `warning` status when they cannot online fetch data. Toggle airplane mode back off to see the warning disappear.

Author: ```suggestion
Disconnect from the network and pan around the map to see the warning status of the different layers when they cannot online fetch data. Reconnect to the network to see the warning disappear.
```
Update from design changes.

## How it works

1. Create an `AGSMap` with some operational layers.
2. Set the map on an `AGSMapView`.
3. Assign a closure to `layerViewStateChangedHandler` of the map view, which get executed every time a layer's view status changes.
4. Get the `AGSLayer` and the current view `status` of type `AGSLayerViewState` defining the new state.

## Relevant API

* AGSLayer
* AGSLayerViewState
* AGSMap
* AGSMapView

## About the data

The [Satellite (MODIS) Thermal Hotspots and Fire Activity](https://runtime.maps.arcgis.com/home/item.html?id=b8f4033069f141729ffb298b7418b653) layer presents detectable thermal activity from MODIS satellites for the last 48 hours. [MODIS Global Fires](https://earthdata.nasa.gov/earth-observation-data/near-real-time/firms/active-fire-data) is a product of NASA’s [Earth Observing System Data and Information System (EOSDIS)](https://earthdata.nasa.gov/about), part of NASA's [Earth Science Data](https://science.nasa.gov/earth-science/earth-data/). EOSDIS integrates remote sensing and GIS technologies to deliver global MODIS hotspot/fire locations to natural resource managers and other stakeholders around the World.

## Additional information

> The following are members of the `AGSLayerViewState.status` enum:

Reviewer: ```suggestion
The following are members of the `AGSLayerViewState` enum:
```
